### ISSUE TITLE
How can I reset the zoom?

### ISSUE THREAD

**Issue by forethoughtde** [anvaka/panzoom#69]
I'd like to reset the zoom size of the element to initial size. Can I know if there is any built-in method provides this functionality. Otherwise, how can I achieve this result? 

Thanks in advance

**Comment by shivarajalagond** [comment]
You can do something like this to restet,

`$("#ID").panzoom("disable");`  -- Where ID is the DOM element you used to zoomin

**Comment by GxCreations** [comment]
I use this in my project:

`instance.zoomAbs(
                0, // initial x position
                0, // initial y position
                1  // initial zoom 
            );`

**Comment by manishofficial** [comment]
use this function and call for input button
<input type="button" value="Reset" onclick="div_reset();" />

function div_reset(){
  panzoom(document.querySelector('.zoom'), {
  }).zoomAbs(
    0
  );
}

**Comment by anilkumar1994** [comment]
This works fine with zooming. But the position of the image changes.

**Comment by vijilov** [comment]
I am facing the similar issue. 

The reset works fine, however the image/svg does not Zoom-in/ Zoom-out from the default(initial) position.

Here is my code that I have used,

```
var area = document.getElementById('zoomPanCont');
var instance = "";

instance  = panzoom(area, { 
    bounds: true,
    maxZoom: 1.55,
    minZoom: .8,
    zoomDoubleClickSpeed: 1
});

function resetPanZoom(){
	$(this).instance = panzoom(area, { 
    bounds: true,
    maxZoom: 1.55,
    minZoom: .8,
    zoomDoubleClickSpeed: 1
    }).zoomAbs( 0, 0, 1 );
}


$("#btnResetZoom").on("click",function(){
    resetPanZoom();
});
$("#btnZoomIn").on("click",function(){
    instance.smoothZoom(0, 0, 1.25);
});
$("#btnZoomOut").on("click",function(){
    instance.smoothZoom(0, 0, 0.8);
});
```

Can you help me with a snippet how to add a reset function, and the ZoomIn function?

**Comment by anvaka** [comment]
TL;DR: 

``` js
function resetZoom(panZoomInstance) {
  panZoomInstance.moveTo(0, 0);
  panZoomInstance.zoomAbs(0, 0, 1);
}
```

Zoom (or scaling operation) always needs transform origin (a single point that remains unchanged after scale is done). For this reason when you call `zoomAbs(x, y, scale)` we take `(x, y)` coordinates. These coordinates are given in client space of the container (not your scene). So `(0, 0)` would always be top left corner of the `container`, not of the content that you are trying to zoom.

To make sure the transform is also reset to its initial state, we call `.moveTo(0, 0)` which pans directly to the origin of the content (top left corner of your content), after which we perform `zoomAbs(0, 0, 1)`.

@vijilov  in your demo you don't need to recreate multple panzoom instances, since they would be fighting with each other for user events. Just use the `moveTo()/zoomAbs()` trick as described. Here is a fixed demo: https://jsfiddle.net/myxbv9qd/

**Comment by vijilov** [comment]
Thank you for your help. It is working.

 Is it possible to give smooth transition effect on reset also like zoomin & zoomout?

**Comment by NCenerar** [comment]
Hello. I was just trying to get a reset to original state on double click and face the same issue :
Reset the zoom and go to original coordinate.
The @anvaka solution with two command was not satisfying because it was not "pretty".
Anyway, here is the solution I found :

```
const xys = panzoom.getTransform()
// panzoom is the reference to the panzoom object
if(xys.scale > 1) { // calculate the point that should not move
    const fScale = 1 - xys.scale
    const fixeX = xys.x / fScale
    const fixeY = xys.y / fScale
    panzoom.smoothZoomAbs(fixeX, fixeY, 1)
} else { // just go back to (0, 0) and scale 1
    panzoom.moveBy(-xys.x, -xys.y, true)
    panzoom.smoothZoomAbs(xys.x, xys.y, 1)
}
panzoom.moveTo(0, 0) // to be sure to go back to (0,0)
panzoom.zoomAbs(0, 0, 1) // to be sure to go back to scale 1
```

**Comment by wandyatmono** [comment]
This is my solution. hope it would help

```javascript
window.addEventListener('load', function () {
	// The `g4` element is inside svg with `viewBox` and
	// it has the initial `transform`. Panzoom should still work fine.
	var thePZ = document.getElementById('presentation');
	var emdPanzoom = panzoom(thePZ, {
		smoothScroll: true,
		bounds: true
	})
	var origin = emdPanzoom.getTransform(); 
	var x = parseFloat(origin.x.toString(10));
	var y = parseFloat(origin.y.toString(10));
	var scale = parseFloat(origin.scale.toString(10));
	$("#init").touch({
		noClick: false
	});

	$("#init").on("tap", function(event) {
		console.log(x, y, scale);
		emdPanzoom.moveTo(x, y);
		emdPanzoom.zoomAbs(x, y, scale);
	});

	gsap.to("#cross", 1, {
		rotation:360, ease:Linear.easeNone, repeat:0,
		xPercent: 0, //para centrar na linha
		yPercent: 0, //para centrar na linha
		transformOrigin: "50% 50%"
	})
});
```

**Comment by g-otn** [comment]
@NCenerar solution is the TL;DR for smooth reset of the pan/zoom. Just make sure the `bounds` option is disabled.

**Comment by mruac** [comment]
Hi all, I have found an api endpoint named `showRectangle()`, to which I am not totally sure what it does but I have found it great way to replace the `zoomAbs()` method as it seems to only work when the element is zoomed in, but not zoomed out.

```js
function resetzoom(instance, container_element){
instance.showRectangle( container_element.getBoundingClientRect() ); instance.moveTo(0,0);
}
```

In my case my panzoom element is a child of `<body\>`, so `container_element` is the `body` element.

The `showRectangle()` doesn't apply the new transforms, so the `moveTo()` applies it.
